controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -184,6 +185,45 @@ public static StreamTruncationRecord computeTruncationRecord(final byte[] indexT
                 previousTruncationRecord.getDeletedSegments(), ImmutableSet.copyOf(toDelete));
     }
 
+    public static long computeSizeTill(final byte[] indexTable, final byte[] historyTable, final byte[] segmentTable,
+                                       final Map<Integer, Long> streamCut, final SealedSegmentsRecord sealedSegmentsRecord) {
+        Preconditions.checkNotNull(streamCut);
+        Preconditions.checkNotNull(indexTable);
+        Preconditions.checkNotNull(historyTable);
+        Preconditions.checkNotNull(sealedSegmentsRecord);
+        Preconditions.checkNotNull(segmentTable);
+        Preconditions.checkArgument(!streamCut.isEmpty());
+        Map<Integer, Integer> epochCutMap = computeEpochCutMap(historyTable, indexTable, segmentTable, streamCut);
+        Map<Segment, Integer> cutMapSegments = transform(segmentTable, epochCutMap);
+
+        AtomicLong size = new AtomicLong();
+
+        Map<Integer, Long> sealedSegmentSizeMap = sealedSegmentsRecord.getSealedSegmentsSizeMap();
+
+        // add sizes for segments in stream cut
+        streamCut.forEach((key, value) -> size.addAndGet(value));
```

Comment:
To confirm, are these the segments that are currently open? I'm asking the because I'm wondering how `SealedSegmentsRecord` works for streams with a fixed number of segments.

Reply:
The purpose of SealedSegmentsRecord is to record the size of a segment at the time when it is sealed. 
Note: this size is absolute size and not the truncated size. Which means it corresponds to total number of bytes written to the segment. 
For a stream with fixed number of segments, SealedSegmentsRecord will be empty (unless a manual scale operation is performed on it).

Reply:
Ok